### DISCUSSION TITLE
List with commits is not complete, not sure if this is a bug

### DISCUSSION THREAD

**Discussion by jpbroeders** [rfennell/AzurePipelines#1432] [Q&A]
Hi,

I have a situation where I have a pullRequest object with some associatedCommits, and one of those associatedCommits isn't in the commits array which is on the same level of the pullRequests.

In my release notes I need to search for every associatedCommit in the seperate list of commits for more detailed data about the path.
I'm not sure if I'm doing something wrong, but in Azure DevOps everything seems ok. I also use the testconsole and enabnle all extra options in the settings, but still I can't match one of the associatedCommits with an item in the commits array.

Is it maybe a bug? Or do I something wrong?

**Comment by rfennell** [comment]
I doubt you are doing anything wrong, but there might be an issue with the say the commits are associated with the PR for your use case.

The logic used is simply a match of commit IDs

```js
 allPullRequests.forEach(pr => {
    if (pr.lastMergeCommit) {
        if (pr.lastMergeCommit.commitId === commit.id) {
             agentApi.logInfo(`- PR ${pr.pullRequestId} matches the commit ${commit.id}`);
                   globalPullRequests.push(<EnrichedGitPullRequest>pr);
        }
   } else {
      agentApi.logInfo(`- PR ${pr.pullRequestId} does not have a lastMergeCommit`);
 });
```

**Reply by jpbroeders** [reply]
Today the same situation occured, and I know what happend. The was a branch created based on our MAIN branch. Via a pull request this commit was pushed into MAIN. Then the person figured out that the change should not be merged into our MAIN branch, but into our RELEASE branch. (we have 2 permanent branches). 

So the person created a new branch based on our release branch and cherry picked the change from the MAIN branch and commit it into the new created branch. This was pushed via a pull request into our release branch. And again the commit information isn't available in the commits list. Only in the associatedCommits list, which is part of the pullRequest object.

**Reply by rfennell** [reply]
I guess the question becomes 'can we PR/WI association code' be altered to match this use case?

Or is it enough to know the process/cause?

**Reply by jpbroeders** [reply]
You are right with the first, can associated commits from other PR (add via cherry pick) be added
